### ISSUE TITLE
Problems with many circuits with one main

### ISSUE THREAD

**Issue by erdoganishe** [dl-solarity/hardhat-zkit#41]
### hardhat-zkit version

0.4.6

### What happened?

I have many circom files with the same main component, but with different compile params.

In result, after compiling then with `npx hardhat zkit compile` command I have next situation:

first circuits files (template name which called by main, index) generated in core directory, others are in generated folders for them. They generated in right way, but everyone except first has own folder for autogen files, and first one generated them right into core directory.

example:
```bash
core
├── RegisterIdentityBuilder.ts
├── index.ts
├── registerIdentity_1_256_3_5_576_248_NA.circom
│   ├── RegisterIdentityBuilder.ts
│   └── index.ts
└── registerIdentity_1_256_3_6_576_248_1_2432_5_296.circom
    ├── RegisterIdentityBuilder.ts
    └── index.ts
```

### Minimal reproduction steps

Create many main components for one template with different params.
Edit hardhat config 
npx hardhat zkit compile

**Comment by Arvolear** [comment]
Hey @erdoganishe, thank you for opening this issue. Could you please elaborate on the bug and provide the directory structure of your circuits?

**Comment by erdoganishe** [comment]
hardhat config:
```ts
...
zkit: {
    circuitsDir: "test/circuits/",
    compilationSettings: {
      c: true,
      onlyFiles: ["registerIdentity_1_256_3_4_600_248_1_1496_3_256.circom","registerIdentity_1_256_3_5_576_248_NA.circom","registerIdentity_1_256_3_6_576_248_1_2432_5_296.circom"],
      skipFiles: []
    },
    setupSettings: {
      ptauDir: "zkit/ptau",
      onlyFiles: ["registerIdentity_1_256_3_4_600_248_1_1496_3_256.circom","registerIdentity_1_256_3_5_576_248_NA.circom","registerIdentity_1_256_3_6_576_248_1_2432_5_296.circom"],
      skipFiles: []
    },
  },
  ...
  ```
  
  directory:
  
  ```bash
  .
├── registerIdentity_1_256_3_4_600_248_1_1496_3_256.circom
├── registerIdentity_1_256_3_5_576_248_NA.circom
└── registerIdentity_1_256_3_6_576_248_1_2432_5_296.circom

```

First circuit (`registerIdentity_1_256_3_4_600_248_1_1496_3_256.circom`) after compiling circuits(`npx hardhat zkit compile`) creates autogen files in core directory, but all other generates a folder, where they have autogen files:

```bash
.
├── RegisterIdentityBuilder.ts
├── index.ts
├── registerIdentity_1_256_3_5_576_248_NA.circom
│   ├── RegisterIdentityBuilder.ts
│   └── index.ts
└── registerIdentity_1_256_3_6_576_248_1_2432_5_296.circom
    ├── RegisterIdentityBuilder.ts
    └── index.ts
```

The are the same except compile params in circuit, so everyone should be in folder, i guess.

**Comment by Arvolear** [comment]
This makes sense. Will look into that.

**Comment by Arvolear** [comment]
This fix will be included in the next major plugin version `0.5.0` together with the support for plonk.
